### PR TITLE
Remove forcing NO_SKID on unsuspecting CRL users

### PR DESCRIPTION
**Pull request by kaleb-himes**

### CODE REVIEW

File: wolfssl/wolfcrypt/settings.h [hunk 1]
```diff
@@ -1621,7 +1621,7 @@ extern void uITRON4_free(void *p) ;
 #ifdef HAVE_CRL
     /* not widely supported yet */
     #undef NO_SKID
```
Comment (toddouska) [inline wolfssl/wolfcrypt/settings.h]: I'm not sure this undef should stay in.  Silently removing potential user settings is not ideal.

File: wolfcrypt/src/asn.c [hunk 2]
```diff
@@ -16026,15 +16034,17 @@ int ParseCRL(DecodedCRL* dcrl, const byte* buff, word32 sz, void* cm)
...
 
     /* openssl doesn't add skid by default for CRLs cause firefox chokes
-       we're not assuming it's available yet */
-#if !defined(NO_SKID) && defined(CRL_SKID_READY)
-    if (dcrl->extAuthKeyIdSet)
-        ca = GetCA(cm, dcrl->extAuthKeyId);
+       if experiencing issues uncomment NO_SKID define in CRL section of
+       wolfssl/wolfcrypt/settings.h */
+#ifndef NO_SKID
+    ca = GetCAByName(cm, dcrl->crlHash); /* most unique */
```
Comment (toddouska) [inline wolfcrypt/src/asn.c]: Is this the correct first one?

File: src/ssl.c [hunk 3]
```diff
@@ -4583,6 +4583,10 @@ Signer* GetCAByName(void* vp, byte* hash)
             if (XMEMCMP(hash, signers->subjectNameHash,
                         SIGNER_DIGEST_SIZE) == 0) {
                 ret = signers;
+            } else if (cm->crl != NULL && cm->crl->crlList != NULL &&
+                       XMEMCMP(hash, cm->crl->crlList->crlHash,
```
Comment (toddouska) [inline src/ssl.c]: Why would the first CRL_Entry on the list always be the correct one?

File: wolfssl/wolfcrypt/settings.h [hunk 4]
```diff
@@ -1619,9 +1619,9 @@ extern void uITRON4_free(void *p) ;
 #endif
 
 #ifdef HAVE_CRL
-    /* not widely supported yet */
-    #undef NO_SKID
-    #define NO_SKID
+    /* may not be widely supported */
```
[the remark this answers is not in the document]
Comment (ejohnstown) [reply wolfssl/wolfcrypt/settings.h]: Agreed. Don't change this.

File: wolfssl/wolfcrypt/asn.h [hunk 5]
```diff
@@ -1345,11 +1345,13 @@ struct DecodedCRL {
     byte    crlHash[SIGNER_DIGEST_SIZE]; /* raw crl data hash            */
     byte    lastDate[MAX_DATE_SIZE]; /* last date updated  */
     byte    nextDate[MAX_DATE_SIZE]; /* next update date   */
+    byte    extAuthKeyId[KEYID_SIZE]; /* Authority Key ID                */
     byte    lastDateFormat;          /* format of last date */
     byte    nextDateFormat;          /* format of next date */
     RevokedCert* certs;              /* revoked cert list  */
     int          totalCerts;         /* number on list     */
     void*   heap;
+    byte extAuthKeyIdSet : 1;      /* Set when the AKID was read from CRL */
```
[the remark this answers is not in the document]
Comment (kaleb-himes) [reply wolfssl/wolfcrypt/asn.h]: it was a copy/paste. No harm in removing the :1, just turns it into a full byte (8-bits) as opposed to a single bit.

**Pull request merged**